Automattic/jetpack
# Tiled Gallery Block: Fix block crashing on high volume gallery (> 50 images).

Fixes https://github.com/Automattic/jetpack/issues/25017 https://github.com/Automattic/wp-calypso/issues/60335 

#### Changes proposed in this Pull Request:
Fixes the Tiled Gallery Block crashing when user updates a high volume gallery (> 50 images), e.g. adding images, remove images or simplying updating without changing anything.

**Why did it break?**
React throws a "maximum call stack depth" error.  This is because of excessive calls to `this.props.setAttributes` causing unnecessary re-rendering.

**How was it fixed?**
Add additional checks to ensure `this.props.setAttributes` is only called when necessary.

#### Jetpack product discussion
Maintenance work.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:

**How to replicate the crash:**
- Add a Tiled Gallery block with more than 50 images.
- Click Edit Gallery from the block toolbar.
- Add images, remove images or just do nothing and click Update in the media modal.
- It should crash.
- Save this post where you can consistently replicate the crash.

**How to test the fix:**
- Load the post where you can consistently replicate that crash.
- Click Edit Gallery from the toolbar.
- Add images, remove images or just do nothing and click Update in the media modal.
- It should NOT crash.

Here are some videos to show the before & after:


https://user-images.githubusercontent.com/1287077/162379721-1d955629-dd8a-4989-8c29-5b52200461c5.mp4


https://user-images.githubusercontent.com/1287077/162379526-a3b2cfcc-02f8-4650-a476-09db4fa4a94f.mp4
